exporters/sender/okhttp/src/main/java/io/opentelemetry/exporter/sender/okhttp/internal/RetryInterceptor.java

```diff
@@ -88,31 +90,31 @@ public Response intercept(Chain chain) throws IOException {
         if (response != null) {
           response.close();
         }
+        exception = null;
       }
-
-      attempt++;
       try {
         response = chain.proceed(chain.request());
+        if (response != null) {
```

Comment:
What's the motivation for changing this part of the logic?

Reply:
1. for better readability, it's easier to read in depth than in length
2. if response is null and no exception happened code fails in `throw exception;` line because exception is null. when response if null it's not something transient

Reply:
I suppose this is possible but I haven't seen response null in practice. If it does occur, we can simply add a null check immediately after `response = chain.proceed(chain.request())`.

Reply:
it's exactly what I did in this PR 😃 
also null check was before this change so I internationally keep it (but I suspect I can just drop it)

also previous code has the issue - if previous(before last) attempt returned rertryable response but the last attempt gets retryable exception method still returns the previous response which is not good as the last state should be returned